### PR TITLE
https_proxy fix for download behind a web proxy on CI machines

### PR DESCRIPTION
**Pull request by broekema41**
Dear jfrog team. 

This fix will allow CI machines that only have access to the internet over a enterprise proxy to use the npm package.

If you have questions please do not hesitate to ask. I will post a temporary npm package on jfrog-cli-go-proxy. 

kind regards,
Roland

### CODE REVIEW

File: npm/init.js [hunk 1]
```diff
@@ -16,15 +17,36 @@ function validateNpmVersion() {
     }
 }
 
+function redirectDetectDownload(starturl) {
+    if(process.env.https_proxy && process.env.https_proxy.length > 0) {
+        var mainurl = process.env.https_proxy + starturl.replace('https://', '/https/');
+        console.log(mainurl);
+        http.get(mainurl, function(res) {
+            if(res.statusCode == 302) {
```
Comment (eyalbe4) [inline npm/init.js]: I suggest changing this if statement to handle both 301 and 302 status codes:
ֿֿ```if (res.statusCode == 301 || res.statusCode == 302) {```

File: npm/init.js [hunk 2]
```diff
@@ -16,15 +17,36 @@ function validateNpmVersion() {
     }
 }
 
+function redirectDetectDownload(starturl) {
```
Comment (eyalbe4) [inline npm/init.js]: I suggest renaming this function to ```download(url)```

File: npm/init.js [hunk 3]
```diff
@@ -16,15 +17,36 @@ function validateNpmVersion() {
...
+        }).on('error', function (err) {console.error(err);});
+    } else {
+        https.get(starturl, function(res) {
+            if(res.statusCode == 302) {
+                redirectDetectDownload(res.headers.location);
+            } else if (res.statusCode == 200) {
+                writeToFile(res)
+            } else {
+                console.log('Unexpected status code during JFrog CLI download')
+            }
```
Comment (eyalbe4) [inline npm/init.js]: The above code lines seem to be duplicated. Can we put them in a function, which we can invoke twice?

**Comment by eyalbe4** [comment]
Thank you for this contribution @broekema41!
See my inline comments.
Can you also create this PR on the *dev* branch? (See the *Guidelines* section under *Pull requests* in the README).
Will you please join our contributors community by signing [JFrog's CLA](https://secure.echosign.com/public/hostedForm?formid=5IYKLZ2RXB543N) so that we can merge your code? 
Thanks

**Comment by broekema41** [comment]
thank you [eyalbe4](/eyalbe4) i have simplified the code as per your comments. 
will close this pull request and re-open to your dev branch

**Pull request closed without merging**